lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/FunctionalChaosResult.cxx

```diff
@@ -300,6 +301,63 @@ void FunctionalChaosResult::load(Advocate & adv)
   adv.loadAttribute( "composedMetaModel_", composedMetaModel_ );
 }
 
+IndicesCollection FunctionalChaosResult::getIndicesHistory() const
+{
+  if (metaModel_.getOutputDimension() > 1)
+    throw NotYetImplementedException(HERE) << "getIndicesHistory is only available for 1-d output dimension";
+  return IndicesCollection(indicesHistory_);
+}
+
+Collection<Point> FunctionalChaosResult::getCoefficientsHistory() const
+{
+  if (metaModel_.getOutputDimension() > 1)
+    throw NotYetImplementedException(HERE) << "getCoefficientsHistory is only available for 1-d output dimension";
+  return coefficientsHistory_;
+}
+
+void FunctionalChaosResult::setSelectionHistory(Collection<Indices> & indicesHistory, Collection<Point> & coefficientsHistory)
+{
+  indicesHistory_ = indicesHistory;
+  coefficientsHistory_ = coefficientsHistory;
+}
 
+Graph FunctionalChaosResult::drawSelectionHistory() const
+{
+  if (metaModel_.getOutputDimension() > 1)
+    throw NotYetImplementedException(HERE) << "drawSelectionHistory is only available for 1-d output dimension";
+  const UnsignedInteger size = indicesHistory_.getSize();
+  if (!size)
+    throw InvalidArgumentException(HERE) << "No selection history available";
+
+  // compute union of basis terms
+  std::map <UnsignedInteger, UnsignedInteger> indicesMap;
```

Comment:
Do you need a map here? Or is an unordered_map ok? It is more efficient than a map.

Reply:
unordered_map is ok too